README.md
# DCG.Altis | Changelog
20-9-2015 : TandelK
---

1. Added TAW View Distance Script to the mission file .


16 Sept 2015: TandelK
---

1. Added Restore Gear after respawn script
2. Added JIP Wait script for checking if player is connected properly or not

15 Sept 2015 : xBloodshed
---
1. fixed Arsenal cleanup bug
2. removed unnecessary decoratives

15 Sept 2015 : Iniyan
---
1. Fixed the TFAR long radio issue. This should fix the backpacks issue.
2. Support for RHS chinooks and MV22 added to igiLoad.
3. TFAR default frequencies set to 123.3, 100.1, 101.1 on channels 1,2&3. LR frequencies 50, 60 on channels 1&2.
4. Changed mission file version to v1.11

14 Sept 2015 : Iniyan
---
1. Fixed backpacks dropped while setting loadouts. Also fixed the loadout resetting when somebody joins/exits the server.
2. Pilot restriction fixed.
3. TFAR frequencies set to ITGC frequencies.

13 Sept 2015 : Iniyan
---
1. Custom virtual arsenal (CVA) script optimized. Need to choose weapons and individual items per class.
2. Added folder class names for dev purposes. Delete it later. Contains all weapons & optics, mags, items (ACE3, RHS, vanilla) in the required format for CVA.
3. Pilot restrictions imposed. Both the pilot classes (Heli & Jet) can fly both types. Other roles are kicked out.

13 Sept 2015 : TandelK
---
1. Added cTab to Officers 
2. Added file called **Classnames.txt** containig all the classnames required for the mission
3. Changed Helmets for every units to UCP 

13 Sept 2015 : xBloodshed
---
1. Added save Gear module (ace)
2. Minor base cleanup and changed vehicles to RHS
3. Changed roles as following:
 1. Platoon Commander
   * Platoon JTAC
    * Platoon Medic
    * Platoon Engineer
 2. Squad Lead
   * Squad Engineer
 3. Fireteam Lead
   * Fireteam Auto Rifleman
    * Marksman
    * AT Rifleman
    * Corpsman
    * Rifleman Light
4. Changed mission load name to **DCG for ITGC**
5. Changed all units to USArmy UCP and removed eventHandler from all. 
```javascript
this addeventhandler ["respawn","_this execVM 'scripts\ITGC_Loadouts\lo_xxx.sqf'"];
```

12 Sept 2015: Iniyan
---
1.	Added custom loadouts based on roles. Loadouts are at folder **_"scripts\ITGC_Loadouts"_**. I've added some loadouts on my preference. Review them please.
	Some loadouts like the engineer and JTAC are not complete. I based the loadouts on our current count of players and situations we face in daily missions.
	Some loadouts like the AR and Ammo bearer are very heavy. Edit them as you see fit.
2.	Removed vanilla items/weapons from vehicles spawned at vehicle centre. Added Medical stuff, rifle/MG ammo, smokes, AT & AA rounds instead.
	Edited file _"fnc\fn_spawnVVS.sqf"_. Pls review the numbers and edit them as fit.
3.	Removed "Take Radio" option available at the arsenal.


Changelog.txt
12 Sept 2015: Iniyan
-------------

1.	Added custom loadouts based on roles. Loadouts are at folder "scripts\ITGC_Loadouts". I've added some loadouts on my preference. Review them please.
	Some loadouts like the engineer and JTAC are not complete. I based the loadouts on our current count of players and situations we face in daily missions.
	Some loadouts like the AR and Ammo bearer are very heavy. Edit them as you see fit.

2.	Removed vanilla items/weapons from vehicles spawned at vehicle centre. Added Medical stuff, rifle/MG ammo, smokes, AT & AA rounds instead.
	Edited file "fnc\fn_spawnVVS.sqf". Pls review the numbers and edit them as fit.

3.	Removed "Take Radio" option available at the arsenal.

13 Sept 2015 : TandelK
-------------

1.  Added cTab to Officers 

2. Added file called Classnames.txt containig all the classnames required for the mission

3. Changed Helmets for every units to UCP 

13 Sept 2015 : Iniyan
-------------

1.	Custom virtual arsenal (CVA) script optimized. Need to choose weapons and individual items per class.

2.	Added folder class names for dev purposes. Delete it later. Contains all weapons & optics, mags, items (ACE3, RHS, vanilla) in the required format for CVA.

3.	Pilot restrictions imposed. Both the pilot classes (Heli & Jet) can fly both types. Other roles are kicked out.

16 Sept 2015: TandelK
-------------

1. Added Respawn Gear script 



data/SEN_params.hpp
class param0 {
	title = "---------------------- Dynamic Combat Generator Parameters ----------------------";
	values[] = {1};
	texts[] = {""};
	default = 1;
   SEN_paramCode = "";
};
class param1 {
   title = "Debug Mode";
   values[] = {0,1};
   texts[] = {"Off", "On"};
   default = 0;
   SEN_paramCode = "";
};
class param2 {
   title = "Disable Third Person Camera";
   values[] = {0,1};
   texts[] = {"Off", "On"};
   default = 0;
   SEN_paramCode = "";
};
class param3 {
   title = "Time of Day";
   values[] = {-1,0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23};
   texts[] = {"Random","00:00","01:00","02:00","03:00","04:00","05:00","06:00","07:00","08:00","09:00","10:00","11:00",
   "12:00","13:00","14:00","15:00","16:00","17:00","18:00","19:00","20:00","21:00","22:00","23:00"};
   default = -1;
   SEN_paramCode = "";
};
class param4 {
   title = "Enemy Side";
   values[] = {0,1};
   texts[] = {"East", "Independent"};
   default = 0;
   SEN_paramCode = "";
};
class param5 {
   title = "Enemy Patrols";
   values[] = {0,1};
   texts[] = {"Off", "On"};
   default = 1;
   SEN_paramCode = "";
};
class param6 {
   title = "Armored Patrol Count";
   values[] = {0,20,50,80};
   texts[] = {"None", "Low", "Medium", "High"};
   default = 20;
   SEN_paramCode = "";
};
class param7 {
   title = "Occupied Location Count";
   values[] = {1,2};
   texts[] = {"1", "2"};
   default = 2;
   SEN_paramCode = "";
};
class param8 {
   title = "Occupied Location Enemy Count";
   values[] = {20,40};
   texts[] = {"Low", "High"};
   default = 20;
   SEN_paramCode = "";
};
class param9 {
	title = "Tasks Count";
	values[] = {1,2,3,4};
	texts[] = {"1", "2", "3", "4"};
	default = 2;
   SEN_paramCode = "";
};
class param10 {
	title = "Occupied Location Visual Effects";
	values[] = {0,1};
	texts[] = {"Off","On"};
	default = 1;
   SEN_paramCode = "";
};
class param11 {
   title = "Use Modified Content if Available";
   values[] = {0,1};
   texts[] = {"Off","On"};
   default = 1;
   SEN_paramCode = "";
};
class param12 {
   title = "---------------------- ACE 3 Parameters ----------------------";
   values[] = {1};
   texts[] = {""};
   default = 1;
   SEN_paramCode = "";
};
class param13 {
   title = "* requires Advanced Medical Level";
   values[] = {1};
   texts[] = {""};
   default = 1;
   SEN_paramCode = "";
};
class param14 {
   title = "Medical Level";
   values[] = {1,2};
   texts[] = {"Basic","Advanced"};
   default = 2;
   SEN_paramCode = "ace_medical_level = %1;";
};
class param15 {
   title = "Advanced Wounds *";
   values[] = {0,1};
   texts[] = {"Off","On"};
   default = 0;
   SEN_paramCode = "ace_medical_enableAdvancedWounds = %1;";
};
class param16 {
   title = "Player Damage Threshold";
   values[] = {8,12,16};
   texts[] = {"Low","Medium","High"};
   default = 12;
   SEN_paramCode = "ace_medical_playerDamageThreshold = ((%1)*0.1);";
};
class param17 {
   title = "Max Revive Time (minutes)";
   values[] = {60,300,600,900};
   texts[] = {"1","5","10","15"};
   default = 300;
   SEN_paramCode = "ace_medical_maxReviveTime = %1;";
};
class param18 {
   title = "Personal Aid Kit Usage *";
   values[] = {0,1};
   texts[] = {"Anyone","Medic Only"};
   default = 1;
   SEN_paramCode = "ace_medical_medicSetting_PAK = %1;";
};
class param19 {
   title = "Consume Personal Aid Kit *";
   values[] = {0,1};
   texts[] = {"No","Yes"};
   default = 1;
   SEN_paramCode = "ace_medical_consumeItem_PAK = %1;";
};
class param20 {
   title = "Personal Aid Kit Usage Location *";
   values[] = {0,1,2,3,4};
   texts[] = {"Anywhere","Medical Vehicles","Medical Facilities","Medical Vehicles and Facilities","Disabled"};
   default = 0;
   SEN_paramCode = "ace_medical_useLocation_PAK = %1;";
};
class param21 {
   title = "Personal Aid Kit Usage Condition *";
   values[] = {0,1};
   texts[] = {"Usable Any Time","Usable on Stable Patient"};
   default = 1;
   SEN_paramCode = "ace_medical_useCondition_PAK = %1;";
};
class param22 {
   title = "Surgical Kit Usage *";
   values[] = {0,1};
   texts[] = {"Anyone","Medic Only"};
   default = 1;
   SEN_paramCode = "ace_medical_medicSetting_SurgicalKit = %1;";
};
class param23 {
   title = "Consume Surgical Kit *";
   values[] = {0,1};
   texts[] = {"No","Yes"};
   default = 1;
   SEN_paramCode = "ace_medical_consumeItem_SurgicalKit = %1;";
};
class param24 {
   title = "Surgical Kit Usage Location *";
   values[] = {0,1,2,3,4};
   texts[] = {"Anywhere","Medical Vehicles","Medical Facilities","Medical Vehicles and Facilities","Disabled"};
   default = 3;
   SEN_paramCode = "ace_medical_useLocation_SurgicalKit = %1;";
};
class param25 {
   title = "Surgical Kit Usage Condition *";
   values[] = {0,1};
   texts[] = {"Usable Any Time","Usable on Stable Patient"};
   default = 1;
   SEN_paramCode = "ace_medical_useCondition_SurgicalKit = %1;";
};
class param26 {
   title = "Max View Distance (meters)";
   values[] = {3000,6000,9000};
   texts[] = {"3000","6000","9000"};
   default = 6000;
   SEN_paramCode = "ace_viewdistance_limitViewDistance = %1;";
};
class param27 {
   title = "Advanced Ballistics";
   values[] = {0,1};
   texts[] = {"Off","On"};
   default = 1;
   SEN_paramCode = "ace_advanced_ballistics_enabled = %1;";
};